Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/DataEditorForm.java

```diff
@@ -275,19 +282,20 @@ public void validate() {
      * @return navigation target
      */
     public String save() {
-        try {
-            metadataPanel.preserve();
-            structurePanel.preserve();
-            ServiceManager.getProcessService().updateChildrenFromRootElement(process, workpiece.getRootElement());
+        metadataPanel.preserve();
+        structurePanel.preserve();
+        if (validate()) {
             try (OutputStream out = ServiceManager.getFileService().write(mainFileUri)) {
                 ServiceManager.getMetsService().save(workpiece, out);
+                return close();
+            } catch (IOException e) {
+                Helper.setErrorMessage(e.getLocalizedMessage(), logger, e);
```

Comment:
Is it passing of logger inside the error message not equal to log stuff? Is the second logger below needed?